File: OffByOne/src/OffByOne.Ql/Ast/Expressions/Unary/NotExpression.cs
namespace OffByOne.Ql.Ast.Expressions.Unary
{
    using OffByOne.Ql.Ast.Expressions.Unary.Base;
    using OffByOne.Ql.Common.Visitors.Contracts;

    public class NotExpression : UnaryExpression
    {
        public NotExpression(Expression expression)
            : base(expression)
        {
        }

        public override TResult Accept<TResult, TContext>(
            IExpressionVisitor<TResult, TContext> visitor,
            TContext environment)
        {
            return visitor.Visit(this, environment);
        }
    }
}

File: vanVlaanderenDerksen/QLJava/src/sc/ql/model/expressions/NotExpression.java
package sc.ql.model.expressions;

import sc.ql.model.Expression;
import sc.ql.model.visitors.ExpressionVisitor;

public class NotExpression extends Expression {
	private final Expression expression;
	
	public NotExpression(Expression expression) {
		this.expression = expression;
	}
	
	public Expression getExpression() {
		return this.expression;
	}

	@Override
	public <T> T accept(ExpressionVisitor<T> visitor) {
		return visitor.visit(this);
	}
}
